### DISCUSSION TITLE
Export e-field data

### DISCUSSION THREAD

**Discussion by ptrem** [ansys/pyaedt#1973] [Q&A] [answered]
Hi,
I currently set up my first simulation with HFSS and pyaedt.
Therefore I would like to extract the e-field data in defined planes like the coordinate planes.
I created a plot using the functions `create_fieldplot_cutplane` and `plot_field_from_fieldplot`.
Now I would like to extract the field data of these cutplanes for example as numpy array.
I already tried to use the functions `get_efields_data` and `get_far_field_data`, but by using this functions an error occurs. 
Has anyone an idea how to extract the e-field data in the cutplanes like shown in the plot or the full-field e-field data?
(The zip file contains a demo simulation)

Thank you in advance

E-Field Plot
![image](https://user-images.githubusercontent.com/58425593/201112879-20f8fd24-b4a2-451e-a973-8f49f43790d0.png)

Error using `get_efields_data`
![image](https://user-images.githubusercontent.com/58425593/201113703-4546d8ed-2bee-4d9c-b7fb-cbd6da85ffda.png)

Error using `get_far_field_data`
![image](https://user-images.githubusercontent.com/58425593/201115227-0293f546-d344-4e15-9d09-63a969e1d687.png)

[SMA_RF_Jack.zip](https://github.com/pyansys/pyaedt/files/9981573/SMA_RF_Jack.zip)

**Comment by ptrem** [comment]
As far as i found out, there is not a direct method to export this data, but it is possible to extract the field data in a plane by using the 
`export_fild_file_on_grid` method. This method extracts a file with the field data in a grid and returns the file path. This file can be read again and processed into a numpy array or whatever u need.

**Comment by Samuelopez-ansys** [comment, marked as answer]
@ptrem You can use plot_field method which does what you asked:

https://aedt.docs.pyansys.com/version/stable/API/_autosummary/pyaedt.modules.AdvancedPostProcessing.PostProcessor.plot_field.html
